app/lib/project/UFOExportController.es6.js

```diff
@@ -0,0 +1,128 @@
+
+    _p.exportGenerator = function*(){
+        var model = this._project.open(this._masterName)
+          , master = model.query('master#' + this._masterName)
+          , glyphs = master.children
+          , glyphSet
+          , glyph
+          , drawFunc
+          , updatedUFOData
+          , i, l, v, ki, kil, k, keys
+          , style
+          , time, one, total = 0
+          ;
+
+        console.warn('setting up UFO directory structure...');
+
+        // create a bare ufoV2 directory
+        this._io.mkDir(false, this._dirName);
+
+        // create dirName/metainfo.plist
+        this._io.writeFile(false, this._dirName+'/metainfo.plist'
+                                , plistLib.createPlistString(metainfoV2));
+
+        // fontforge requires a fontinfo.plist that defines unitsPerEm
+        this._io.writeFile(false, this._dirName+'/fontinfo.plist'
+                                , plistLib.createPlistString(minimalFontinfo));
+
+        this._io.mkDir(false, this._dirName+'/glyphs');
+        this._io.writeFile(false, this._dirName+'/glyphs/contents.plist', plistLib.createPlistString({}));
+
+        glyphSet = GlyphSet.factory(false, this._io, this._dirName+'/glyphs', undefined, 2);
+
+        console.warn('exporting glyphs...');
+        for(i = 0,l=glyphs.length;i<l;i++) {
+            glyph = glyphs[i];
+            style = model.getComputedStyle(glyph);
+            time = timer.now();
+            drawFunc = glyphBasics.drawGlyphToPointPen.bind(
+                this
+              , {
+                      penstroke: glyphBasics.renderPenstrokeOutline
+                    , contour: glyphBasics.renderContour
+                }
+              , model, glyph);
+
+            // Allow the glyph ufo data to be updated by the CPS.
+            updatedUFOData = glyph.getUFOData();
+            keys = Object.keys(updatedUFOData);
+            for(ki=0,kil=keys.length;ki<kil;ki++) {
+                try {
+                    k = keys[ki];
+                    v = style.get(MOMGlyph.convertUFOtoCPSKey(k));
+                    updatedUFOData[k] = v;
+                }
+                catch( error ) {
+                    if(!(error instanceof errors.Key)) {
+                        throw error;
+                    }
+                }
+            }
+            glyphSet.writeGlyph(false, glyph.id, updatedUFOData, drawFunc,
+                                      undefined, {precision: this._precision});
+            one = timer.now() - time;
+            total += one;
+            console.warn('exported', glyph.id, 'this took', one,'ms');
+            yield {'current_glyph':i, 'total_glyphs':l, 'glyph_id':glyph.id};
+        }
+        console.warn('finished ', i, 'glyphs in', total
+            , 'ms\n\tthat\'s', total/i, 'per glyph\n\t   and'
+            , (1000 * i / total)  ,' glyphs per second.'
+        );
+        glyphSet.writeContents(false);
+    };
+
+    _p.run_export_iteration = function() {
```

Comment:
This method is bad. It hides more than it helps. Especially the `this.gen` property is odd. If you would run this "from outside" and let the caller manage the generator, the intent would be much clearer. Also you wouldn't forget to reset `this.gen` after its done, but that does not matter, because this method should be deleted.

What you should do here instead is bringing back the old "_p.do_export" public interface. This can look very much like `drawGlyphToPointPen` of `rendering/glyphBasics.es6.js` https://github.com/felipesanches/metapolator/blob/improve_progressbar/app/lib/rendering/glyphBasics.es6.js#L219
Just a simple convenince 2-liner, so that we still have it available if we don't want to run it in pieces, for nodeJS  or webworkers for example.

 Maybe we should take the chance and rename it into `doExport` and also rename the counterpart in `OTFExportController` https://github.com/felipesanches/metapolator/blob/improve_progressbar/app/lib/project/OTFExportController.js#L55